File: 2414-length-of-the-longest-alphabetical-continuous-substring/2414-length-of-the-longest-alphabetical-continuous-substring.py
class Solution:
    def longestContinuousSubstring(self, s):
        st = []
        empty = []
        ans = -1
        for ch in s:
            if len(st) == 0 or ( len(st) != 0 and ord(st[-1]) + 1 == ord(ch) ):
                st.append(ch)
            elif ord(st[-1]) + 1 != ord(ch):
                size = len(st)
                ans = max(ans, size)
                st = empty.copy()
                st.append(ch)
        return max(ans,len(st))

File: 2414-length-of-the-longest-alphabetical-continuous-substring/2414-length-of-the-longest-alphabetical-continuous-substring.cpp
class Solution {
public:
    int longestContinuousSubstring(string s) {
        stack<char>st;
        stack<char>empty;
        int ans = -1 ;
        for(char ch : s){
            if(st.empty()){
                st.push(ch);
            }
            if(!st.empty() && int(st.top())+1 == int (ch)){
                st.push(ch);
            }
            else if( int(st.top())+1 != int(ch) ){
                int size = st.size();
                ans=max(ans,size);
                st=empty;
                st.push(ch);
            }
        }
        if(st.size() > ans) return st.size();
        return ans;
    }
};